tests/multi_ptr/multi_ptr_deduction_guides.cpp

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2017-2022 Codeplay Software LTD. All Rights Reserved.
+//  Copyright (c) 2022-2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+//  Provides multi_ptr deduction guides tests
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "../common/get_cts_string.h"
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_deduction_guides {
+using namespace sycl;
+
+template <typename T>
+class check_multi_ptr_deduction {
+ public:
+  void operator()(const std::string& type) {
+    typeName = type;
+
+    check_for_space<global>();
+    check_for_space<local>();
+  }
+
+ private:
+  static constexpr access::address_space global =
+      access::address_space::global_space;
+  static constexpr access::address_space local =
+      access::address_space::local_space;
+
+  std::string typeName;
```

Comment:
Coding style everywhere
```suggestion
  std::string type_name;
```